Lib/test/test_os.py

```diff
@@ -4640,6 +4640,12 @@ class A(os.PathLike):
     def test_pathlike_class_getitem(self):
         self.assertIsInstance(os.PathLike[bytes], types.GenericAlias)
 
+    def test_pathlike_subclass_slots(self):
+        class A(os.PathLike):
+            __slots__ = ()
+            def __fspath__(self):
+                return ''
+        self.assertFalse(hasattr(A(), '__dict__'))
```

Comment:
Does this need to be so fancy, or could you just do `hasattr(os.PathLike.__slots__)`?

Reply:
I think it is testing on purpose that subclasses do inherit the slots definition.

(And you meant `hasattr(os.PathLike, '__slots__')` 😉)

Reply:
If we only check for `__slots__`, we miss cases where a superclass of `os.PathLike` does not define `__slots__`:

```python
>>> class A:
...     pass
... 
>>> class B(A):
...     __slots__ = ()
... 
>>> B.__slots__
()
>>> B().__dict__
{}
```